### ISSUE TITLE
If user is unsubscribed should we even emit pending ?

### ISSUE THREAD

**Issue by sublimator** [coilhq/web-monetization-projects#141]
Noticed after the time-to-pending fix that emitted pending immediately that if you are unsubscribed then pending is emitted, an auth check done, then stopped. 

This causes "unlocking ... " to flicker briefly before "subscribe to never see this annoying box" on coil.com when user is unsubscribed.

We could potentially only emit pending if we are optimistic that our subscription is active, as so:

```typescript
  async startWebMonetization(
    request: StartWebMonetization,
    sender: MessageSender
  ) {
    const tab = getTab(sender)
    this.tabStates.logLastMonetizationCommand(tab, 'start')
    // This used to be sent from content script as a separate message
    this.mayMonetizeSite(sender)
    const userBeforeReAuth = this.store.user
    let emittedPending = false
    const emitPending = () => this.sendSetMonetizationStateMessage(tab, 'pending')

    // If we think we have an active subscription, emit pending immediately
    // otherwise wait until try to auth, then emit
    if (userBeforeReAuth?.subscription?.active) {
      emittedPending = true
      emitPending()
    }

    log('startWebMonetization, request', request)
    const { requestId } = request.data

    log('loading token for monetization', requestId)
    const token = await this.auth.getTokenMaybeRefreshAndStoreState()
    if (!token) {
      // not signed in.
      console.warn('startWebMonetization cancelled; no token')
      this.sendSetMonetizationStateMessage(tab, 'stopped')
      return false
    }
    if (!this.store.user?.subscription?.active) {
      this.sendSetMonetizationStateMessage(tab, 'stopped')
      console.warn('startWebMonetization cancelled; no active subscription')
      return false
    }

    if (!emittedPending) {
      emitPending()
    }

    const lastCommand = this.tabStates.get(tab).lastMonetization.command
    if (lastCommand !== 'start') {
      console.warn('startWebMonetization cancelled via', lastCommand)
      return false
    }

    log('starting stream', requestId)
    this.tabsToStreams[tab] = requestId
    this.streamsToTabs[requestId] = tab
    this.streams.beginStream(requestId, {
      token,
      ...request.data,
      initiatingUrl: request.data.initiatingUrl
    })

    return true
  }
```

**Comment by sharafian** [comment]
I think this is a good idea. We could think about the semantics either as:

- pending occurs until we're sure we can/can't pay, at which point it settles in started/stopped
- stopped occurs until we're sure we can pay, at which point it moves to pending and then started

So this change would be the second state model, right? But where we sometimes instantly know that we're subscribed and can move to pending

**Comment by sublimator** [comment]
> So this change would be the second state model, right?

Ja, we can't really know 100% for sure. They could be sub/unsub since last user request. 
If our latest view of the world means we are optimistic that we are subbed, then emit pending immediately. If our view is that we are unsubbed, then wait in stopped state until we've got latest results, potentially skipping pending entirely.

**Comment by sublimator** [comment]
Closed with #148
